library/src/main/kotlin/build/buf/connect/impl/ProtocolClient.kt

```diff
@@ -52,8 +53,10 @@ class ProtocolClient(
         val serializationStrategy = config.serializationStrategy
         val requestCodec = serializationStrategy.codec(methodSpec.requestClass)
         try {
+            val host = URI(config.host)
```

Comment:
This is still re-parsing the string every time you make a new request. Why do that more than once if the string cannot change?